File: Property-Management-Web/src/core/interfaces/api/CreateTenantRequest.ts
export interface CreateTenantRequest {
  firstName: string
  lastName: string
  email: string
  phoneNumber: string
  propertyId: number
  leaseStartDate?: string
  leaseEndDate?: string
  tenantStatus: 'active' | 'past_due' | 'former' | 'applicant'
}

File: Property-Management-Api/Models/Request/CreateTenantRequest.cs
namespace Property_Management_Api.Models.Request;

public sealed class CreateTenantRequest
{
    public string FirstName { get; init; } = string.Empty;
    public string LastName { get; init; } = string.Empty;
    public string Email { get; init; } = string.Empty;
    public string PhoneNumber { get; init; } = string.Empty;
    public int PropertyId { get; init; }
    public DateTime? LeaseStartDate { get; init; }
    public DateTime? LeaseEndDate { get; init; }
    public string TenantStatus { get; init; } = "active";
}
